straxen/plugins/acqmon_processing.py

```diff
@@ -90,44 +94,122 @@ def setup(self):
+                    # Find the time of stop_j that is closest to time of start_i
+                    inx = np.searchsorted(veto_hits_stop['time'], time, side = 'right') 
+                    
+                    if inx == len(veto_hits_stop['time']):
+                        continue
+                    else:
+                        res['veto_interval'].append(veto_hits_stop['time'][inx] - time)
+                        res["time"].append(time)
+                        res["endtime"].append(veto_hits_stop['time'][inx])
+                        res["veto_type"].append(name + 'veto')
+        
+        result = strax.dict_to_rec(res, self.dtype)
         return result
 
-
 @numba.njit
-def channel_select(rr, ch_stop, ch_start):
-    """Return data in between stop and start channels of the acquisition monitor (AM)"""
-    return rr[((rr['channel'] == ch_stop) | (rr['channel'] == ch_start))] 
+def channel_select_(rr, ch):
+    """Return data from start/stop veto channel in the acquisition monitor (AM)"""
+    return rr[rr['channel'] == ch] 
+
+
+@export
+@strax.takes_config(
+    strax.Option('veto_proximity_window', default=int(5e8), type=int, track=True,
+                 help='Maximum separation between veto stop and start pulses [ns]'))
+class VetoProximity(strax.OverlapWindowPlugin):
+    """ 
+    Find the closest next/previous veto start and end to each event center.
+    
+    previous_x: Time in ns between the time center of an event and the previous x
+    next_x: Time in ns between the time center of an event and the next x
+    This also considers any x inside the event. x could be either:
+        - busy_x: busy on/off signal
+        - he_x:   high energy channels busy on/off signal
+        - hev_x:  high energy veto on/off signal
+    """
+        
+    __version__ = '0.0.2'
+    depends_on = ('events', 'veto_intervals')
+    provides = ('veto_proximity')
+    data_kind = ('events')
+    
+    veto_names = ['busy', 'he', 'hev']
+    
+    save_when = strax.SaveWhen.NEVER
+    
+    def infer_dtype(self):
+        dtype = []
+        for n in self.veto_names:
+            dtype.append((('Time to previous %s veto start [ns]' %n, 'previous_%s_on' %n), np.int64))
+            dtype.append((('Time to previous %s veto end [ns]' %n, 'previous_%s_off' %n), np.int64))
+            dtype.append((('Time to next %s veto start [ns]' %n, 'next_%s_on' %n), np.int64))
+            dtype.append((('Time to next %s veto end [ns]' %n, 'next_%s_off' %n), np.int64))
+        
+        dtype += strax.time_fields
+
+        return dtype
+    
+    def setup(self):
+        self.states = ['on', 'off']
+    
+    def get_window_size(self):
+        return (self.config['veto_proximity_window'] * 100)
+        
+    def compute(self, events, veto_intervals):
+        res = defaultdict(list)
+        
+        t_event_centers = (events['time'] + events['endtime'])//2
+
+        for n, name in enumerate(self.veto_names):
```

Comment:
I would have used here a different strategy but many roads lead to Rom. If we run into performance issues as Joran already mentioned one could use something like: 
```
touching = strax.touching_windows(vetos, events_with_an_extended_time_and_endtime)
....
```